Counter/src/app/shared/pokemon-service.ts

```diff
@@ -1,16 +1,37 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Observable, of, map } from 'rxjs';
+import { Observable, of, map, BehaviorSubject, switchMap } from 'rxjs';
 import { Pokemon } from './pokemon';
 import { POKEMON } from './pokemon-list';
 import { ImageData } from './image';
 
 @Injectable()
 export class PokemonService {
-    constructor(private http:HttpClient) { }
+    private filterStream: BehaviorSubject<string> = new BehaviorSubject("");
+    public filter: Observable<string>;
+    public pokemon: Observable<Pokemon[]>;
+    private filteredStream: BehaviorSubject<boolean> = new BehaviorSubject(false);
```

Comment:
Something to consider... instead of managing another stream just to know if it has been filtered, you could use the pokemon observable to send back a more complex object that returns the pokemon as well as the criteria that it was filtered by. It could grow into telling you what page, what sort, etc.